OpenRA.Game/Support/ConditionExpression.cs

```diff
@@ -0,0 +1,901 @@
+			{
+				foreach (var t in postfix)
+				{
+					switch (t.Type)
+					{
+						case TokenType.And:
+						{
+							BooleanJunction(t.Type);
+							continue;
+						}
+
+						case TokenType.Or:
+						{
+							BooleanJunction(t.Type);
+							continue;
+						}
+
+						case TokenType.NotEquals:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.NotEqual(x, y));
+							continue;
+						}
+
+						case TokenType.Equals:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.Equal(x, y));
+							continue;
+						}
+
+						case TokenType.Not:
+						{
+							if (ast.PeekType() == ExpressionType.Bool)
+								ast.Push(Expressions.Expression.Not(ast.Pop(ExpressionType.Bool)));
+							else
+								ast.Push(AsNegBool(ast.Pop(ExpressionType.Int)));
+							continue;
+						}
+
+						case TokenType.OnesComplement:
+						{
+							ast.Push(Expressions.Expression.OnesComplement(ast.Pop(ExpressionType.Int)));
+							continue;
+						}
+
+						case TokenType.LessThan:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.LessThan(x, y));
+							continue;
+						}
+
+						case TokenType.LessThanOrEqual:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.LessThanOrEqual(x, y));
+							continue;
+						}
+
+						case TokenType.GreaterThan:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.GreaterThan(x, y));
+							continue;
+						}
+
+						case TokenType.GreaterThanOrEqual:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.GreaterThanOrEqual(x, y));
+							continue;
+						}
+
+						case TokenType.Add:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.Add(x, y));
+							continue;
+						}
+
+						case TokenType.Subtract:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.Subtract(x, y));
+							continue;
+						}
+
+						case TokenType.Multiply:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							ast.Push(Expressions.Expression.Multiply(x, y));
+							continue;
+						}
+
+						case TokenType.Divide:
+						{
+							var y = ast.Pop(ExpressionType.Int);
+							var x = ast.Pop(ExpressionType.Int);
+							var isNotZero = Expressions.Expression.NotEqual(y, Zero);
+							var divide = Expressions.Expression.Divide(x, y);
+							ast.Push(IfThenElse(isNotZero, divide, MaxInt));
```

Comment:
I have two concerns with returning MaxInt here:
* Dividing a negative number by 0 discards the sign
* Integer overflow means that an expresion like `x / total < 10` is true, but `x / total + 1 < 10` is *false* if total == 0.

Returning 0 would be better,  because we're effectively returning NaN as a falsey value.